src/main/java/com/thomashan/coup/player/Player.java

```diff
@@ -0,0 +1,110 @@
+package com.thomashan.coup.player;
+
+import com.thomashan.coup.action.MainActionType;
+import com.thomashan.coup.card.Card;
+import com.thomashan.coup.card.PlayerCard;
+import com.thomashan.coup.card.PlayerCards;
+import lombok.ToString;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@ToString
+public final class Player {
+    private final int coins;
+    private final PlayerCards playerCards;
+
+    private Player(int coins, Card card1, Card card2) {
+        this.coins = coins;
+        this.playerCards = PlayerCards.of(card1, card2);
+    }
+
+    private Player(int coins, PlayerCards playerCards) {
+        this.coins = coins;
+        this.playerCards = playerCards;
+    }
+
+    public static Player of(Card card1, Card card2) {
+        return new Player(2, card1, card2);
+    }
+
+    public Player revealCard(Card card) {
+        return new Player(getCoins(), playerCards.reveal(card));
+    }
+
+    public Player minus(Card card) {
+        return new Player(getCoins(), getPlayerCards().minus(PlayerCard.of(card)));
+    }
+
+    public Player plus(Card card) {
+        return new Player(getCoins(), playerCards.plus(PlayerCard.of(card)));
+    }
+
+    public boolean isActive() {
+        return playerCards.isActive();
+    }
+
+    public int getCoins() {
+        return coins;
+    }
+
+    public PlayerCards getPlayerCards() {
+        return playerCards;
+    }
+
+    public List<Card> getActiveCards() {
+        return playerCards.getActiveCards();
+    }
+
+    public Set<Card> getActiveCardSet() {
+        return playerCards.getActiveCardSet();
+    }
+
+    public Player income() {
+        return new Player(getCoins() + 1, playerCards);
+    }
+
+    public Player foreignAid() {
+        return new Player(getCoins() + 2, playerCards);
+    }
+
+    public Player tax() {
+        return new Player(getCoins() + 3, playerCards);
+    }
+
+    public Player useCoinsForAssassination() {
+        return new Player(getCoins() - 3, playerCards);
+    }
+
+    public Player stealFromOtherPlayer(int otherPlayerCoins) {
+        if (otherPlayerCoins < 2) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using Literals in Conditional Statements](https://app.codacy.com/app/thomashan/coup-java/pullRequest?prid=4561385)